Assets/_Game/Scripts/Lighting/LightFlicker.cs

```diff
@@ -13,11 +13,10 @@ public class LightFlicker : MonoBehaviour
     public float radiusIncrementPerUpdate;
     public float intensityIncrementPerUpdate;
     public int framesPerLightUpdate;
-    public string lightName;
 
-    void Start()
+    void Awake()
     {
-        flickerLight = GameObject.Find(lightName).GetComponent<Light2D>();
+        flickerLight = GetComponentInChildren<Light2D>();
```

Comment:
Using the name of the game object to retrieve a reference to a component can be useful, but should be avoided if possible. This is especially true when relying on a manually entered string value from the inspector.

Since this script is attached to a prefab, and we know that the Light2D component we are looking for is within it, we can use a recursive search down to child game objects. You could also manually assign the reference to the component in the inspector. I went with the recursive search in this case, although it could be improved with a null reference check.

Another point, using `Awake` instead of `Start` in this case would be advised. Here are some links to information regarding the 2 fixes:
https://docs.unity3d.com/ScriptReference/Component.GetComponentsInChildren.html
https://gamedevbeginner.com/start-vs-awake-in-unity/

Reply:
Yeah that makes sense, I knew there was a way to do this properly, I just found the issue when I made a prefab and dropped a couple of lights in which meant only one of them was being updated by the script. Thanks for the information!